### PR TITLE
analyze using the latest sdk from head

### PR DESCRIPTION
**Pull request by devoncarew**
- analyze using the latest sdk from head
- some general maintenance to prefer calling `dart pub` to `pub`

This changes the analysis in ci/analyze.sh to analyze against the newly built dart sdk from head. Previously, we'd been (perhaps unintentionally) using the checked in dart sdk. This brings the analysis in-line with what lib/web_ui uses as well.

### CODE REVIEW

File: ci/analyze.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ function follow_links() (
 SCRIPT_DIR=$(follow_links "$(dirname -- "${BASH_SOURCE[0]}")")
 SRC_DIR="$(cd "$SCRIPT_DIR/../.."; pwd -P)"
 FLUTTER_DIR="$SRC_DIR/flutter"
-DART_BIN="$SRC_DIR/third_party/dart/tools/sdks/dart-sdk/bin"
+DART_BIN="$SRC_DIR/out/host_debug_unopt/dart-sdk/bin"
```
Comment (zanderso) [inline ci/analyze.sh:33]: Check that this exists, and fail with an error message that gives some build instructions, like:
```
$ gn --runtime-mode debug --unopt --no-lto
$ ninja -C out/host_debug_unopt
```

Comment (devoncarew) [reply ci/analyze.sh:33]: Gotcha. I have `'gn --unoptimized --runtime-mode=debug'` from the compiling-the-engine wiki, but am seeing:

```
ERROR No command specified.
Most commonly you want "gn gen <out_dir>" to make a build dir.
```

Is there a build target I should specify for the gn command?

Comment (zanderso) [reply ci/analyze.sh:33]: Oh, sorry. The engine tree has it's own script that wraps `gn`. It's under `flutter/tools/gn`.

**Comment by devoncarew** [comment]
ptal, updated based on review feedback

**Review by zanderso — APPROVED** [review]

**Pull request merged**